src/psyclone/psyir/nodes/acc_directives.py

```diff
@@ -254,7 +225,10 @@ def lower_to_language_level(self):
         # TODO GOcean grid properties are duplicated in this set under
         # different names (the OpenACC deep copy support should spot this).
         for pdir in self._acc_dirs:
-            self._sig_set.update(pdir.kernel_references)
+            try:
+                self._sig_set.update(pdir.kernel_references)
+            except TypeError:
+                self._sig_set.update(*pdir.kernel_references)
```

Comment:
Please add a comment here explaining why we have this try-catch.

Reply:
I was kinda hoping you'd tell me not to do this and instead change the return type of `kernel_references` to either a 1-tuple or a 2-tuple of sets rather than the current set or 2-tuple of sets... I didn't do that at the first instance because it always looks a bit weird to return a 1-tuple but...

Reply:
If you were already thinking it would be better done a different way then I change my mind and request that you do that :-)

Reply:
Done. 😝

Reply:
Thanks - I think you were right; it's nicer now.